airflow/models/__init__.py

```diff
@@ -1400,6 +1403,14 @@ def signal_handler(signum, frame):
 
                 task_copy.post_execute(context=context, result=result)
 
+                end_time = time.time()
+                duration = end_time - start_time
+                Stats.timing(
+                    '{dag_id}.{task_id}.duration'.format(
```

Comment:
Do we need to worry about non-safe characters in the dag or task id, or maximum length?

Reply:
I think that worth another PR to fix it.

Reply:
https://github.com/apache/airflow/pull/4889